### PR TITLE
[14.0][IMP][payroll]Add the company inprove in multi-company environment

### PR DESCRIPTION
**Pull request by percevaq**
In multi-company environments you need to filter payroll processing by company.

### CODE REVIEW

**Comment by percevaq** [comment]
Can you please give me a link to the modifications?

Thank you

File: payroll/__manifest__.py
Comment (mtelahun) [inline payroll/__manifest__.py]: Please remove this. It will be done automatically when the PR is merged.

File: payroll/models/hr_payslip_run.py
Comment (mtelahun) [inline payroll/models/hr_payslip_run.py:39]: Please also add the accompanying xml view field properly restricted to users with multi-company access. There are plenty of examples in other modules if you need help.

**Review by mtelahun — CHANGES_REQUESTED** [review]
Please take care of the problems flagged by pre-commit.

File: payroll/__manifest__.py
Comment (percevaq) [reply payroll/__manifest__.py]: Following the oca guide, if you change the structure of the object, you have to make a version change in the y... but I don't know if you have to keep the z.

File: payroll/models/hr_payslip_run.py
Comment (percevaq) [reply payroll/models/hr_payslip_run.py:39]: Yes, right to fix it.

Thanks for your comments.

File: payroll/__manifest__.py
Comment (mtelahun) [reply payroll/__manifest__.py]: When the ocabot merges your changes it will bump the version as necessary. You don't have to worry about this.

**Comment by nimarosa** [comment]
@percevaq  Yes please don't bump the version. It will automatically done by the merge bot. 

Apply the changes suggested by @mtelahun and we merge.

**Comment by ghost** [comment]
Dear @percevaq,
Thank you for the contribution with the company_id field!

To finish this PR, we should do 4 changes:

* Remove the change of version
* In hr_payslip_run_views.xml, add `<field name="company_id" groups="base.group_multi_company" />`
* In hr_payslip_run.py, use double-quotes instead of single-quotes:
```
        comodel_name="res.company",
        string="Company",
```
* Squash all changes into 1 commit
`git rebase -i HEAD~2` will let you squash two commits together. Replace `pick` with `squash` on the second line.

`git commit -a --amend` will add the new changes to the last commit (different commit id)

@percevaq Would you like to do finish the PR?

**Comment by nimarosa** [comment]
@percevaq Are you going to finish this PR?

**Comment by percevaq** [comment]
Sorry for the delay in my reply and thank you for your comments. I think I have it all sorted out.

Best regards

**Comment by nimarosa** [comment]
@percevaq hello, since PRs has been merged before yours you have to do a git rebase to the base branch. 

Also you still need to fix your changes to make pre-commit error don't show up. Look at checks and then you can see why pre-commit is failing and the changes that you have to make. 

Please also first do the rebase since we can't merge this if no based on last branch version.

**Comment by nimarosa** [comment]
Closed in favour of #93 which adds this field.

**Pull request closed without merging**
